tasks/main.yml

```diff
@@ -17,26 +12,328 @@
+        - '"ipaddress" not in __mssql_custom_ip_address.stdout'
+        - '"tcpport" in __mssql_custom_tcp_port.stdout'
+
+    - name: >-
+        Check the password matches the existing password.
+        When the IP address and TCP port settings are custom.
+      command: >-
+        /opt/mssql-tools/bin/sqlcmd
+        -S {{ __mssql_custom_ip_address.stdout |
+        regex_search('[^: ]*$') }},{{ __mssql_custom_tcp_port.stdout |
+        regex_search('[^: ]*$') }}
+        -U sa -P '{{ mssql_password }}' -Q 'SELECT @@VERSION'
+      register: __mssql_sqlcmd_query_custom_settings
+      ignore_errors: true
+      changed_when: false
+      when:
+        - '"ipaddress" in __mssql_custom_ip_address.stdout'
+        - '"tcpport" in __mssql_custom_tcp_port.stdout'
+
+    - name: Ensure that the mssql-server service is stopped
+      service:
+        name: mssql-server
+        state: stopped
+      when: >-
+        ((__mssql_sqlcmd_query_default_settings.skipped is not defined) and
+        (__mssql_sqlcmd_query_default_settings.rc != 0)) or
+        ((__mssql_sqlcmd_query_custom_ip.skipped is not defined) and
+        (__mssql_sqlcmd_query_custom_ip.rc != 0)) or
+        ((__mssql_sqlcmd_query_custom_tcp_port.skipped is not defined) and
+        (__mssql_sqlcmd_query_custom_tcp_port.rc != 0)) or
+        ((__mssql_sqlcmd_query_custom_settings.skipped is not defined) and
+        (__mssql_sqlcmd_query_custom_settings.rc != 0))
+      notify: Restart the mssql-server service
+
+    - name: Change the password of the sa user
+      command: /opt/mssql/bin/mssql-conf set-sa-password
+      environment:
+        - MSSQL_SA_PASSWORD: "{{ mssql_password }}"
+      when: >-
+        ((__mssql_sqlcmd_query_default_settings.skipped is not defined) and
+        (__mssql_sqlcmd_query_default_settings.rc != 0)) or
+        ((__mssql_sqlcmd_query_custom_ip.skipped is not defined) and
+        (__mssql_sqlcmd_query_custom_ip.rc != 0)) or
+        ((__mssql_sqlcmd_query_custom_tcp_port.skipped is not defined) and
+        (__mssql_sqlcmd_query_custom_tcp_port.rc != 0)) or
+        ((__mssql_sqlcmd_query_custom_settings.skipped is not defined) and
+        (__mssql_sqlcmd_query_custom_settings.rc != 0))
+      notify: Restart the mssql-server service
+
+- name: Set a new edition for MSSQL Server
+  when:
+    - __mssql_conf_setup.skipped is defined
+    - mssql_edition is defined
+  block:
+    - name: Register the content of the errorlog file in a custom location
+      command: >-
+        head {{ __mssql_errorlogfile.stdout | regex_search('[^: ]*$') }}
+      register: __mssql_errorlog_custom_content
+      changed_when: false
+      when: "'errorlogfile' in __mssql_errorlogfile.stdout"
+
+    - name: Register the content of the errorlog file in the default location
+      command: head /var/opt/mssql/log/errorlog
+      register: __mssql_errorlog_default_content
+      changed_when: false
+      when: "'errorlogfile' not in __mssql_errorlogfile.stdout"
```

Comment:
See above - use `head {{ __mssql_errorlogfile.stdout }}` and `when __mssql_errorlogfile.stdout | length > 0`